src/lib.rs

```diff
@@ -285,11 +285,27 @@ impl Iterator for MailEntries {
     }
 }
 
+
+/// Invalid subfolder name
+#[derive(Debug)]
+pub struct InvalidFolderName;
+
+impl fmt::Display for InvalidFolderName {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Invalid Folder Name")
+    }
+}
+
+impl std::error::Error for InvalidFolderName {
+}
+
+
 #[derive(Debug)]
 pub enum MaildirError {
     Io(std::io::Error),
     Utf8(std::str::Utf8Error),
     Time(std::time::SystemTimeError),
+    Naming(InvalidFolderName),
```

Comment:
I'd prefer to just make this enum variant `InvalidFolderName(std::string::String)`, where the string stored is the invalid name that was provided.